### PR TITLE
Update pipeline for PHP 8 / 8.1

### PR DESCRIPTION
**Pull request by fritz-gerneth**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,7 +36,7 @@ unit-ci: unit
 	if [ -f ./build/logs/clover.xml ]; then wget https://scrutinizer-ci.com/ocular.phar && sleep 3 && php ocular.phar code-coverage:upload --format=php-clover ./build/logs/clover.xml; fi
 
 infection:
-	$(DOCKER_RUN) vendor/bin/infection --ansi --min-msi=100 --min-covered-msi=100 --threads=$(nproc)
```
Comment (fritz-gerneth) [inline Makefile:39]: Removed the min thresholds for now as the current tests fail these

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -30,13 +30,13 @@ jobs:
   qa:
     strategy:
       matrix:
```
Comment (WyriHaximus) [inline .github/workflows/ci.yml:33]:
```suggestion
      fail-fast: false
      matrix:
```

File: Makefile (hunk 1 above)
Comment (WyriHaximus) [reply Makefile:39]: Will have a look at that after I merged this PR like with the metric package 👍

**Review by WyriHaximus — CHANGES_REQUESTED** [review]
Great work so far, one tiny change request to make reviewing it easier

**Comment by fritz-gerneth** [comment]
All green now

**Review by WyriHaximus — APPROVED** [review]

**Pull request merged**